optuna/samplers/_nsgaiii.py

```diff
@@ -86,12 +86,12 @@ def __init__(
         swapping_prob: float = 0.5,
         seed: int | None = None,
         constraints_func: Callable[[FrozenTrial], Sequence[float]] | None = None,
+        reference_points: np.ndarray | None = None,
+        dividing_parameter: int = 3,
```

Comment:
Nit: we have this many arguments, so how about making these parameters keyword-only?

Reply:
We had a discussion and we decided that it is okay to break the existing interface and make the arguments keyword-only.

Reply:
I updated the arguments in the same way as  https://github.com/optuna/optuna/pull/4806/commits/fd761a7dcfb45d280baea1873e54923c01f0fdd8. 
Could you take another look?